src/Catel.MVVM/Collections/ObservableDictionary.cs

```diff
@@ -100,593 +63,107 @@ public ObservableDictionary(IDictionary<TKey, TValue> dictionary)
+        /// <remarks>Set <see cref="FastObservableDictionary{TKey,TValue}.AutomaticallyDispatchChangeNotifications"/> to <c>false</c> if you do not wish to dispatch to the UI.</remarks>
         /// <param name="action">The <see cref="NotifyCollectionChangedAction"/> operation that was executed.</param>
         /// <param name="newItem">The new item.</param>
         /// <param name="oldItem">The old item.</param>
         /// <param name="index">The index value of the old item affected.</param>
         protected virtual void OnCollectionChanged(NotifyCollectionChangedAction action, KeyValuePair<TKey, TValue> newItem, KeyValuePair<TKey, TValue> oldItem, int index)
         {
-            OnPropertyChanged();
-
-            if (AutomaticallyDispatchChangeNotifications)
-            {
-                _dispatcherService.Value.BeginInvokeIfRequired(() => CollectionChanged?.Invoke(this, new NotifyCollectionChangedEventArgs(action, newItem, oldItem, index)));
-            }
-            else
-            {
-                CollectionChanged?.Invoke(this, new NotifyCollectionChangedEventArgs(action, newItem, oldItem, index));
-            }
-        }
-
-        private void OnPropertyChanged()
-        {
-            OnPropertyChanged(nameof(Count));
-            OnPropertyChanged("Item[]");
-            OnPropertyChanged(nameof(Keys));
-            OnPropertyChanged(nameof(Values));
+            base.OnCollectionChanged(new NotifyCollectionChangedEventArgs(action, newItem, oldItem, index));
         }
 
         /// <summary>
         /// Raises the <see cref="INotifyPropertyChanged.PropertyChanged"/> event while also ensuring that it is dispatched to the UI thread.
         /// </summary>
-        /// <remarks>Set <see cref="AutomaticallyDispatchChangeNotifications"/> to <c>false</c> if you do not wish to dispatch to the UI.</remarks>
+        /// <remarks>Set <see cref="FastObservableDictionary{TKey,TValue}.AutomaticallyDispatchChangeNotifications"/> to <c>false</c> if you do not wish to dispatch to the UI.</remarks>
         /// <param name="propertyName">The target property to invoke <see cref="INotifyPropertyChanged.PropertyChanged"/> on.</param>
         protected virtual void OnPropertyChanged(string propertyName)
         {
-            if (AutomaticallyDispatchChangeNotifications)
-            {
-                _dispatcherService.Value.BeginInvokeIfRequired(() => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName)));
-            }
-            else
-            {
-                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
-            }
+            base.OnPropertyChanged(new PropertyChangedEventArgs(propertyName));
         }
 
 #if NET || NETCORE
         [Serializable]
+        [ObsoleteEx(RemoveInVersion = "6.0", TreatAsErrorFromVersion = "5.12.1", Message = "Use the normal enumerator")]
```

Comment:
Perfect

Reply:
should i mark the `ObservableDictionary` class as obsolete too ?

Reply:
Yeah, I think that's a good idea. Let's keep things consistent and call it FastObservableDictionary (same as FastObservableCollection).